### DISCUSSION TITLE
AzureOpenAI and model

### DISCUSSION THREAD

**Discussion by hjbaard** [RageAgainstThePixel/com.openai.unity#285] [Q&A] [answered]
I'm a bit confused. When I create an implementation in Azure OpenAI Studio I must give up a name (deploymentId) and a Modelname and some more settings but than I can call the api like this and give a different model:
```
var chatRequest = new ChatRequest(messages, Model.GPT4o);
var response = await api.ChatEndpoint.GetCompletionAsync(chatRequest);
```

My question is. Is this valid? Or do I have to give the same model is defined in the Azure implementation?

**Comment by StephenHodgson** [comment]
>  do I have to give the same model is defined in the Azure implementation?

yes. give it the same model as the deployment.

**Reply by hjbaard** [reply]
No I did not, and that is my point. What happens if the model in the implementation is different from the one, I use in the api call.

**Reply by StephenHodgson** [reply]
I am unsure. I suppose it is discarded. You can test and find out. I do not have an Azure subscription to try.

**Reply by hjbaard** [reply]
It looks like it's working but I got 429 errors fast. But concluding from your answers (thanks btw!) The model's should always be the same so you don't have unwanted/unpredicted results right?

**Reply by StephenHodgson** [reply, marked as answer]
I believe so, although, I am unsure if you can just skip providing the model altogether. If so, I can just skip adding it as parameter to the request, but need verification before making this change.

**Reply by StephenHodgson** [reply]
@hjbaard can you do a simple curl request and omit the model and test if that works?

**Reply by hjbaard** [reply]
I've tested it with Unity. Provide a different model to the API then configured in the implementation of Azure. And although it looks like it's using the model that I send using the API call I'm not sure what Azure does with other settings. For instance, the Frequency limit (RPM) setting. This setting depends on all the implementations with the same model for the Azure account. (it can fire 429 errors). 

Is that what you ment?
